File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while(!s.equals("1")){
            int len=s.length();
            s=s.replace("0","");
            answer[1]+=(len-s.length());
            answer[0]++;
            StringBuilder tmp = new StringBuilder();
            for(int i=s.length();i!=0;i=i/2){
                tmp.append(i%2);
            }
            s=tmp.reverse().toString();
        }
        
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = [0,0]
    
    while s!='1':
        x=''
        for i in range(len(s)):
            if s[i]=='1':
                x+=s[i]

        n=len(x)
        x=''
        answer[1]+=len(s)-n
        while n!=0:
            x+=str(n%2)
            n=n//2
        s=x[::-1]
        answer[0]+=1
    
    return answer